File: p2pnet/android/app/src/main/java/com/example/p2pnet/ui/login/LoginUiState.kt
package com.example.p2pnet.ui.login

import com.example.p2pnet.ui.Page
import com.example.p2pnet.ui.TabItem

data class LoginUiState(
    val useWss: Boolean = false,
    val serverHost: String = "deepstack.tech",
    val serverPort: String = "10000",
    val username: String = "test",
    val password: String = "test",
    val targetUsername: String = "",
    val loading: Boolean = false,
    val isConnected: Boolean = false,
    val isLoggedIn: Boolean = false,
    val loggedInUsername: String = "",
    val error: String? = null,
    val messages: List<MessageItem> = emptyList(),
    val tabs: List<TabItem> = listOf(TabItem.main()),
    val currentTabIndex: Int = 0,
    val currentPage: Page = Page.Main
)

File: p2pnet/ios/p2pnet/UI/Login/LoginUiState.swift
import Foundation
import Combine

struct LoginUiState: Equatable {
    var useWss: Bool = false
    var serverHost: String = "deepstack.tech"
    var serverPort: String = "10000"
    var username: String = "test"
    var password: String = "test"
    var targetUsername: String = ""
    var loading: Bool = false
    var isConnected: Bool = false
    var isLoggedIn: Bool = false
    var loggedInUsername: String = ""
    var error: String? = nil
    var messages: [MessageItem] = []
    var tabs: [TabItem] = [TabItem.main(), TabItem.wireGuard()]
    var currentTabIndex: Int = 0
    var currentPage: Page = .main
}